learn/ai_powered_search/rest_response_request.mdx

```diff
@@ -0,0 +1,337 @@
+
+The prompt corresponds to the data that the provider will use to generate your document embeddings. Its specific name changes depending on the provider you chose. In Mistral, this is the `input` field. In Cloudflare, it's called `text`.
+
+Most providers accept either a string or an array of strings. A single string will generate one request per document in your database:
+
+```json
+{
+  "EMBEDDER_NAME": {
+    "source": "rest",
+    "url": "PROVIDER_URL",
+    "request": {
+      "model": "MODEL_NAME",
+      "input": "{{text}}"
+    }
+  }
+}
+```
+
+`{{text}}` indicates Meilisearch should replace the contents of a field with your document data, as indicated in the embedder's [`documentTemplate`](/reference/api/settings#documenttemplate).
+
+An array of strings allows Meilisearch to send up to 10 documents in one request, reducing the number of API calls to the provider:
+
+```json
+{
+  "EMBEDDER_NAME": {
+    "source": "rest",
+    "url": "PROVIDER_URL",
+    "request": {
+      "model": "MODEL_NAME",
+      "input": ["{{text}}", "{{..}}"]
+    }
+  }
+}
+```
+
+When using array prompts, the first item must be `{{text}}`. If you want to send multiple documents in a single request, the second array item must be `{{..}}`.
+
+When using other tools, `input` might be called something else, like `text` or `prompt`:
+
+```json
+{
+  "EMBEDDER_NAME": {
+    "source": "rest",
+    "url": "PROVIDER_URL",
+    "request": {
+      "model": "MODEL_NAME",
+      "text": "{{text}}"
+    }
+  }
+}
+```
+
+### Provide other request fields
+
+You may add as many fields to the `request` object as you need. Meilisearch will send them as configured when it queries the embeddings provider.
+
+For example, Mistral allows you to optionally configure an `encoding_format`. Set it by declaring this field in your embedder's `request`:
+
+```json
+{
+  "EMBEDDER_NAME": {
+    "source": "rest",
+    "url": "PROVIDER_URL",
+    "request": {
+      "model": "MODEL_NAME",
+      "input": ["{{text}}", "{{..}}"],
+      "encoding_format": "float"
+    }
+  }
+}
+```
+
+## The embedding response
+
+You must indicate where Meilisearch can find the document embeddings in the provider's response. Consult your provider's API documentation, paying attention to where it places the embeddings.
+
+Cloudflare's embeddings are located in an array inside `response.result.data`. Describe the full path to the embedding array in your embedder's `response`. The first array item must be `"{{embedding}}"`:
+
+```json
+{
+  "EMBEDDER_NAME": {
+    "source": "rest",
+    "url": "PROVIDER_URL",
+    "request": {
+      "text": "{{text}}"
+    },
+    "response": {
+      "result": {
+        "data": ["{{embedding}}"]
+      }
+    }
+  }
+}
+```
+
+If the response contains multiple embeddings, use `"{{..}}"` as its second value:
+
+```json
+{
+  "EMBEDDER_NAME": {
+    "source": "rest",
+    "url": "PROVIDER_URL",
+    "request": {
+      "model": "MODEL_NAME",
+      "input": "{{text}}"
```

Comment:
```suggestion
      "input": ["{{text}}", "{{..}}"]
```

for the response to contains multiple embeddings, the request needs to contain multiple embeddings, so both the request and response must use the `"{{..}}"` repeat placeholder syntax.

Meilisearch will actually enforce this and return an error if only one of the request or response contains the repeat placeholder syntax.

Reply:
Oh, I didn't know we returned an error for mismatched placeholder existence across request/response. Thanks for letting me know.

Reply:
Yes, sample error messages:

- Error while generating embeddings: user error: in `response`: `response` has multiple embeddings, but `request` has only one text to embed
- Error while generating embeddings: user error: in `response`: `response` has a single embedding, but `request` has multiple texts to embed